### ISSUE TITLE
Deprecate ScalarSpaceEncoder

### ISSUE THREAD

**Issue by oxtopus** [numenta/nupic#2004]
`nupic.encoders.scalarspace.ScalarSpaceEncoder()` is redundant with `nupic.swarming.permutationhelpers.PermuteEncoder()` and doesn't meet coding standards.  It should be removed from:

```
nupic/encoders/__init__.py
nupic/encoders/multi.py
nupic/frameworks/opf/exp_generator/ExpGenerator.py
tests/integration/nupic/opf/expgenerator_test.py
tests/swarming/nupic/swarming/experiments/delta/description.py
tests/swarming/nupic/swarming/experiments/delta/permutations.py
```

The scalarspace module should raise a `DeprecationWarning` exception via `warnings.warn()` (https://docs.python.org/2/library/warnings.html#warnings.warn) informing the user that it should no longer be used.

Eventually, the module should be removed entirely.

**Comment by scottpurdy** [comment]
`PermuteEncoder` is a little bit different than the `ScalarSpaceEncoder`. The former will try including and excluding a specific encoder. If you want to try two, it will try each combination (neither, one, the other, or both). The latter let's you try one or the other, but not both. I think this is fine, just want to be explicit that there is a slight difference.

**Comment by oxtopus** [comment]
Neither are encoders, per se.  `ScalarSpaceEncoder` is implemented as an encoder subclass, but its constructor returns one of `DeltaEncoder` or `AdaptiveScalarEncoder`, depending on the arguments, and never an instance of `ScalarSpaceEncoder` itself.  `PermuteEncoder` also is not an encoder, but is a `PermuteVariable` subclass used in swarming.

I don't think there's good use-case for `ScalarSpaceEncoder` as it is, since there's no advantage over calling either of the `DeltaEncoder` or `AdaptiveScalarEncoder` constructors directly.

**Comment by scottpurdy** [comment]
Yeah, if we really need the functionality for swarming then we can have a `PermuteChoice` or something. And even that probably isn't necessary when we can just use PermuteEncoder
